### PR TITLE
Add short differ

### PR DESCRIPTION
**Pull request by SpacePossum**
closes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/2760

### CODE REVIEW

File: tests/RuleSetTest.php [hunk 1]
```diff
@@ -494,14 +492,14 @@ private function sort(array &$data)
     }
 
     /**
-     * @param array $values
+     * @param array $data
      *
      * @return bool
      */
-    private function allInteger(array $values)
+    private function allIntegerKeys(array &$data)
```
Comment (SpacePossum) [inline tests/RuleSetTest.php]: prevents reference mismatch

File: src/Console/ConfigurationResolver.php [hunk 2]
```diff
@@ -253,7 +255,21 @@ public function getConfigFile()
     public function getDiffer()
     {
         if (null === $this->differ) {
-            $this->differ = false === $this->options['diff'] ? new NullDiffer() : new SebastianBergmannDiffer();
+            if (false !== $this->options['differ']) {
+                // 'diff' value does not matter here
```
Comment (SpacePossum) [inline src/Console/ConfigurationResolver.php]: ?

File: src/Console/ConfigurationResolver.php [hunk 3]
```diff
@@ -253,7 +255,21 @@ public function getConfigFile()
...
+                if ('sbd' === $this->options['differ']) {
```
Comment (SpacePossum) [inline src/Console/ConfigurationResolver.php]: what does `long` mean, of which format?

**Comment by SpacePossum** [comment]
>I think this PR should target master (new feature, even if it was already introduced in 2.2 first).

I like more votes on this, but like you said, it was on 2.2 and I see nothing changed that require master now.

>Also, this brings some unrelated cleanups that should be part of a new PR (which could target 2.2).

there is no need for it IMHO

File: src/Console/ConfigurationResolver.php (hunk 2 above)
Comment (julienfalque) [inline src/Console/ConfigurationResolver.php]: If `--differ` option is passed a value then the tool behaves as if `--diff` was passed as well, even if it wasn't.

File: src/Console/ConfigurationResolver.php (hunk 3 above)
Comment (julienfalque) [inline src/Console/ConfigurationResolver.php]: Yeah not very good too... `full-files`?

File: src/Console/ConfigurationResolver.php (hunk 2 above)
Comment (SpacePossum) [inline src/Console/ConfigurationResolver.php]: yes, that is in the README as `default`

File: src/Console/ConfigurationResolver.php (hunk 3 above)
Comment (SpacePossum) [inline src/Console/ConfigurationResolver.php]: you won't _always_ get the `full` file diff even when using the `long` format. 
You get an extended diff in Unified Format (well, on 1.14.1 I hope, before that the format is a bit more loose, i.e. the diff is in a format that can be described as the `sbd` diff. format)

File: src/Console/ConfigurationResolver.php (hunk 2 above)
Comment (julienfalque) [inline src/Console/ConfigurationResolver.php]: This _default_ means that `--diff` is equivalent to `--diff --differ=sbd`, not that `--differ=sbd` is equivalent to `--diff --differ=sbd`. One could expect that changing the diff type does not implicitly enables diff output.

Comment (SpacePossum) [inline src/Console/ConfigurationResolver.php]: >One could expect that changing the diff type does not implicitly enables diff output.

I doubt that, why change or set a format of something you don't want?

I'm not going back and forward on copy and branch change requests that were not a problem before unless someone points out what relevant changes have happened between than and now.

**Comment by keradus** [comment]
Nice to pick it up ! :)

1. Passing differ on 2.2 lie works for a while (while it was breaking sth else, so it was reverted), thus I'm fine to introduce it again for 2.2 line.

2. On 2.3 line code for resolving differ was modified, please try to reuse it to minimize conflicts.

3. `php-cs-fixer fix --diff=$SHOW_DIFF --differ=sbd` would not work at all...

**Comment by SpacePossum** [comment]
>php-cs-fixer fix --diff=$SHOW_DIFF --differ=sbd would not work at all...

good thing it doesn't :)

**Comment by keradus** [comment]
that's the thing. why not? if one want to efficiently manage `--diff` option via boolean flag of his, why to override it ?
either:
- `--diff` is source of true shall diff be generated or not, if yes - then pick differ from `--differ` param
- `--diff` or `--differ` could be use, but not both at the same time.

now, it's confusing that one option is silently changing another one.

**Comment by SpacePossum** [comment]
`--diff` never was a boolean flag, I opt removing it on 3.x and use `differ` only

**Comment by keradus** [comment]
Then my bad. In that case let `--differ` override `--diff` and let's deprecate `--diff`

**Comment by keradus** [comment]
ping @SpacePossum

File: src/Console/Command/FixCommand.php [hunk 4]
```diff
@@ -134,18 +135,19 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
-                'using-cache' => $input->getOption('using-cache'),
-                'cache-file' => $input->getOption('cache-file'),
-                'format' => $input->getOption('format'),
-                'diff' => $input->getOption('diff'),
+                'rules' => $passedRules,
+                'show-progress' => $input->getOption('show-progress'),
                 'stop-on-violation' => $input->getOption('stop-on-violation'),
+                'using-cache' => $input->getOption('using-cache'),
                 'verbosity' => $verbosity,
-                'show-progress' => $input->getOption('show-progress'),
```
Comment (localheinz) [inline src/Console/Command/FixCommand.php]: 👍 https://plugins.jetbrains.com/plugin/5919-lines-sorter

Comment (SpacePossum) [inline src/Console/Command/FixCommand.php]: yeah I got that one, picked it up from another comment of you :)

**Comment by SpacePossum** [comment]
one more PR (https://github.com/sebastianbergmann/diff/pull/60) and I can continue on this :)

**Comment by SpacePossum** [comment]
needs to be done in another way

**Pull request closed without merging**